pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/LuceneFSTIndexReader.java

```diff
@@ -61,19 +65,27 @@ public MutableRoaringBitmap getDocIds(String searchQuery) {
 
   @Override
   public ImmutableRoaringBitmap getDictIds(String searchQuery) {
-    try {
+    try (InvocationScope span = Tracing.getTracer().createScope(LuceneFSTIndexReader.class)) {
       MutableRoaringBitmap dictIds = new MutableRoaringBitmap();
       List<Long> matchingIds = RegexpMatcher.regexMatch(searchQuery, _readFST);
       for (Long matchingId : matchingIds) {
         dictIds.add(matchingId.intValue());
       }
+      record(span, matchingIds.size());
```

Comment:
This is matching values instead of matching documents

Reply:
yes, it's supposed to be recording the size of the set but is recorded as the number of documents. I will just remove this as it isn't very interesting.

Reply:
removed